com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1649,59 +1665,98 @@ private void SendConnectionRequest()
 
         private IEnumerator ApprovalTimeout(ulong clientId)
         {
-            if (IsServer)
-            {
-                NetworkTime timeStarted = LocalTime;
-
-                //We yield every frame incase a pending client disconnects and someone else gets its connection id
-                while (IsListening && (LocalTime - timeStarted).Time < NetworkConfig.ClientConnectionBufferTimeout && PendingClients.ContainsKey(clientId))
+            // Use the current RTT latency between the client and server (instance relative)
```

Comment:
In my opinion, the proper way would be to take a configurable timeout from the game team and not adjust for latency.

If we fear the gamedevs will forget about latency or be surprised, we can add a fixed amount, say 0.25s. But using computed RTT adds unnecessary complexity and mixes abstractions levels.